java/arcs/android/crdt/ParcelableReference.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.crdt
+
+import android.os.Parcel
+import android.os.Parcelable
+import arcs.android.util.writeProto
+import arcs.core.storage.Reference
+import arcs.core.storage.StorageKeyParser
+
+/** Parcelable version of [Reference]. */
+data class ParcelableReference(override val actual: Reference) : ParcelableReferencable {
+    override fun writeToParcel(parcel: Parcel, flags: Int) {
+        super.writeToParcel(parcel, flags)
+        parcel.writeString(actual.id)
+        parcel.writeString(actual.storageKey.toString())
+        actual.version?.let {
+            parcel.writeBoolean(true)
+            parcel.writeProto(it.toProto())
+        } ?: {
+            parcel.writeBoolean(false)
+        }()
```

Comment:
You can just write a null instead of this boolean. `writeProto` and `readVersionMap` both support nulls

Reply:
sweet, thanks!